scipy/sparse/tests/test_base.py

```diff
@@ -3461,6 +3461,167 @@ def test_argmax(self):
+                X.max(axis=axis, explicit=True).A.flatten(),
+                np.array([8, 0, 28, 38, 48])
+            )
+            assert_array_equal(
+                X.min(axis=axis, explicit=True).A.flatten(),
+                np.array([1, 0, -1, 30, 40])
+            )
+
+        # full matrix
+        D = matrix(np.arange(1, 51).reshape(10, 5))
+        X = self.spmatrix(D)
+
+        for axis in axes_even:
+            assert_array_equal(X.max(axis=axis, explicit=True).A.flatten(),
+                               np.array([46, 47, 48, 49, 50]))
+            assert_array_equal(X.min(axis=axis, explicit=True).A.flatten(),
+                               np.array([1, 2, 3, 4, 5]))
+
+        for axis in axes_odd:
+            assert_array_equal(
+                X.max(axis=axis, explicit=True).A.flatten(),
+                np.array([5, 10, 15, 20, 25, 30, 35, 40, 45, 50])
+            )
+            assert_array_equal(
+                X.min(axis=axis, explicit=True).A.flatten(),
+                np.array([1, 6, 11, 16, 21, 26, 31, 36, 41, 46])
+            )
+
+        # empty matrix
+        D = matrix(np.zeros((10, 5)))
+        X = self.spmatrix(D)
+
+        for axis in axes_even:
+            assert_array_equal(
+                X.max(axis=axis, explicit=True).A.flatten(),
+                np.zeros(5)
+            )
+            assert_array_equal(
+                X.min(axis=axis, explicit=True).A.flatten(),
+                np.zeros(5)
+            )
+
+        for axis in axes_odd:
+            assert_array_equal(
+                X.max(axis=axis, explicit=True).A.flatten(),
+                np.zeros(10)
+            )
+            assert_array_equal(
+                X.min(axis=axis, explicit=True).A.flatten(),
+                np.zeros(10)
+            )
+
+        # zero-size matrices
+        D = np.zeros((0, 10))
+        X = self.spmatrix(D)
+        for axis in axes_even:
+            assert_raises(ValueError, X.min, axis=axis, explicit=True)
+            assert_raises(ValueError, X.max, axis=axis, explicit=True)
+        for axis in axes_odd:
+            assert_array_equal(
+                np.zeros((0, 1)),
+                X.min(axis=axis, explicit=True).A
+            )
+            assert_array_equal(
+                np.zeros((0, 1)),
+                X.max(axis=axis, explicit=True).A
+            )
+
+        D = np.zeros((10, 0))
+        X = self.spmatrix(D)
+        for axis in axes_odd:
+            assert_raises(ValueError, X.min, axis=axis, explicit=True)
+            assert_raises(ValueError, X.max, axis=axis, explicit=True)
+        for axis in axes_even:
+            assert_array_equal(
+                np.zeros((1, 0)),
+                X.min(axis=axis, explicit=True).A
+            )
+            assert_array_equal(
+                np.zeros((1, 0)),
+                X.max(axis=axis, explicit=True).A
+            )
+
+    def test_argmax_explicit(self):
+        D1 = np.array([[-1, 5, 2, 3],
```

Comment:
This test doesn't appear to be exercising the case where a row / column is all zero. We should make sure we cover both of those thoroughly.

Reply:
Done